components/proxy/src/main/kotlin/com/hotels/styx/services/HealthCheckMonitoringService.kt

```diff
@@ -104,18 +98,18 @@ internal class HealthCheckMonitoringService(
     internal fun runChecks(application: String, objectStore: StyxObjectStore<RoutingObjectRecord>) {
         val monitoredObjects = discoverMonitoredObjects(application, objectStore)
                 .map {
-                    val tag = healthStatusTag(it.second.tags).orElse(INACTIVE_TAG)
-                    val health = objectHealthFrom(tag).orElse(null)
-                    Triple(it.first, it.second, health)
+                    val tags = it.second.tags
+                    val objectHealth = objectHealthFrom(stateTagValue(tags), healthcheckTagValue(tags))
+                    Triple(it.first, it.second, objectHealth)
                 }
                 .filter { (_, _, objectHealth) -> objectHealth != null }
```

Comment:
My editor says that `objectHealth != null` is always true. Therefore this `filter` can be removed as useless.

Reply:
(Opinionated comment warning)

I'd consider inlining `discoverMonitoredObjects`. It doesn't contribute much but to hide the selection criteria out of sight. :-)

(... okay some tests would be affected)

Reply:
The tests can be removed, I think. The behaviour of the filter is pretty much self evident, especially now the tag logic is elsewhere and standardised.